hasura/graphql-engine
# console: restore partial support for foreign tables (fix #4714)

### Description
This PR fixes an oversight of #4270.

While the console never had explicit support for foreign tables, they used to be implicitly queried alongside tables and views, as foreign tables do appear in `information_schema.tables`. In #4270, we switched to a more direct approach, explicitly querying `pg_class` instead, and explicitly specifying which types of postgres items we wanted to return: tables, views, and materialized views, effectively dropping support for foreign tables.

This PR fixes this by adding foreign tables to the list of items that the console fetches. However, while this restores a previously lost functionality, the console _still doesn't have explicit support for foreign tables_: they are misclassified as views, and some of the features of the console won't work with them (for instance, the "delete view" (_sic_) button will fail to drop the foreign table).

Adding full support for them in the console is beyond the scope of this PR and will be added to the backlog for triage.

### Changelog


### Affected components


### Steps to test and verify
This PR doesn't have tests. After discussing it internally, we agreed that the ROI of the effort it would take to add them wouldn't be worth it, especially considering foreign tables are only partially supported.

The simplest way to verify this is to do something along those lines, using a running postgres server as its own foreign server:
```
create table foo (id integer, name text);
create extension postgres_fdw;
create server foreign_test
  foreign data wrapper postgres_fdw
  options (host 'localhost', port '$PORT', dbname '$DB');
create user mapping for $USER server foreign_test options (user '$USER');
create foreign table foreign_foo (id integer, value text)
  server foreign_test
  options(schema_name 'public', table_name 'foo');
```
